docs/modules/architecture/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,122 @@
+- `memberLimit` - number of search results returned from member to coordinator
+
+If `topK` denotes number of entries requested, then the following should hold in order for the system to be able to return enough results:
+
+- `partitionLimit * partitionCount >= topK`, `partitionLimit &lt;= topK`
+- `memberLimit * memberCount >= topK`, `memberLimit &lt;= topK`
+
+By default `partitionLimit` and `memberLimit` are equal to `topK` which satisfies the above inequalities but causes processing of much more results than the requested `topK`.
+This improves the overall quality of the results but can have a significant performance overhead because more entries are fetched from each partition of the index and sent between the members.
+
+NOTE: Consider tuning `partitionLimit` based on quality and latency requirements. It should be adjusted alongside number of partitions.
+`memberLimit` is less critical for overall behavior if there are only a few members.
+
+=== Single-stage search
+
+A simplified search algorithm exists which does not perform intermediate aggregation of results on member level.
+It is used if the cluster has only a single member. It can be also enabled using search hint in other cases.
+
+In this case search request is executed in parallel on all partitions (on their owners)
+and partition results are aggregated directly on the coordinator member to produce the final result.
+
+This search algorithm has `partitionLimit` parameter which behaves in the same way as for two-stage search.
+
+== Partition count impact
+
+Number of partitions has big impact on the performance of the vector collection. There are numerous conflicting factors impacting selection of optimal partition count:
+
+- *data ingestion*: more partitions give better parallelism, up to some point, around total number of partition threads in the cluster.
+  Having more partitions than that will not improve ingestion speed significantly.
+- *similarity search*: fewer partitions in general results in better search performance and latency.
+  However, the impact on quality/recall is complicated and depends also on `partitionLimit`.
+- *migration*: very big partitions (in terms of memory size) should be avoided. Note that both metadata, vectors and vector index internal representation count.
+  Current general recommendation for partition size is around 50-100MB per partition, which results in fast migrations and small pressure on heap during migration.
+  However, for vector search use case the partition size may be increased above that limit provided that there is enough heap for migrations (see below).
+- *other data structures*: number of partitions is a cluster-wide setting shared by all data structures. If the needs are vastly different, you might consider creating separate clusters.
+
+NOTE: It is not possible to change number of partitions for an existing cluster.
+
+WARNING: Default value of 271 partitions may result in inefficient vector similarity search.
```

Comment:
```suggestion
[WARNING]
====
* The default value of 271 partitions can result in inefficient vector similarity searches.
We recommend that you tune the number of partitions for use in clusters with vector collections.

* The entire collection partition is migrated simultaneously and cannot be chunked.
If using partitions that are larger than the recommended size, ensure that you have sufficient heap memory to run migrations. The amount of heap memory required is approximately the size of the vector collection partition multiplied by the number of parallel migrations.
To decrease pressure on heap memory, you can decrease the number of parallel migrations using `hazelcast.partition.max.parallel.migrations` and `hazelcast.partition.max.parallel.replications`.
====
```

Reply:
They remain two warnings in the suggested change, it is just that they display in a single box. Multiple warnings one after the other are not a good look; beta or otherwise.

I have added a feed-in line to the two warnings, and change the list to a numbered list to highlight that there are two separate things to note.

I have also made it a CAUTION; here's the difference:

- A Caution advises readers to be careful
- A Warning inform readers of danger, harm, or consequences that exist

@k-jamroz WDYT?

Reply:
`CAUTION` is fine

> The entire collection partition is migrated simultaneously and cannot be chunked.

chunked migration is enabled by `hazelcast.partition.migration.chunks.enabled` and is enabled by default. We do not have a lot of docs about it except in https://docs.hazelcast.com/hazelcast/5.4/system-properties. Chunked migration is implemented eg. for IMaps but is not yet implemented for Vector Collections, but hopefully will be later.

Vector collection currently supports fragmented migration (`hazelcast.partition.migration.fragments.enabled`), which means that each collection is migrated in separate packet/message.

So maybe we can use wording similar to system properties docs:

> The entire collection partition is migrated in one big chunk

Reply:
How about:

The entire collection partition is migrated as a single chunk

?